### PR TITLE
msamprz/fix-datepicker-locale-base

### PR DESCRIPTION
**Pull request by msamprz**
Fix datepicker locale base

### CODE REVIEW

File: src/javascript/app_2/Utils/Date/date-time.js [hunk 1]
```diff
@@ -17,11 +17,9 @@ export const epochToMoment = epoch => moment.unix(epoch).utc();
...
     if (!value) return moment().utc(); // returns 'now' moment object
     if (value instanceof moment && value.isValid() && value.isUTC()) return value; // returns if already a moment object
-    const is_number      = typeof value === 'number';
-    // need to explicitly convert date string to a JS Date object then pass that into Moment
-    // to get rid of the warning: Deprecation warning: moment construction falls back to js Date
-    const formatted_date = moment(new Date(value)).format('YYYY-MM-DD');
-    return is_number ? epochToMoment(value) : moment.utc(formatted_date);
+    if (typeof value === 'number') return epochToMoment(value); // returns epochToMoment() if not a date
+
+    return moment.utc(value);
```
Comment (kellybinary) [inline src/javascript/app_2/Utils/Date/date-time.js]: @msamprz please make sure it doesn't show depreciation warning in the console. and also works across browsers (i think moment.js tend to behaves differently in different browsers)

Comment (msamprz) [reply src/javascript/app_2/Utils/Date/date-time.js]: yep, for sure 👍

File: src/javascript/app_2/Utils/Date/date-time.js [hunk 2]
```diff
@@ -1,5 +1,10 @@
 import moment       from 'moment';
 import { localize } from '_common/localize';
+import ServerTime from '../../../_common/base/server_time';
```
Comment (negarn) [inline src/javascript/app_2/Utils/Date/date-time.js]: alignment

File: src/javascript/app_2/App/Components/Form/TimePicker/dialog.jsx [hunk 3]
```diff
@@ -48,7 +48,10 @@ const Dialog = ({
                                 start_time_moment,
                                 end_time_moment,
                                 'minute');
-                            const is_enabled = is_hour_enabled && is_minute_enabled;
+                            const last_interval_of_hour = 52;
```
Comment (negarn) [inline src/javascript/app_2/App/Components/Form/TimePicker/dialog.jsx]: can you add a comment about where this comes from?

File: src/javascript/app_2/Utils/Date/date-time.js [hunk 4]
```diff
@@ -1,5 +1,10 @@
...
+
+moment.createFromInputFallback = function (config) {
+    config._d = new Date(NaN);
```
Comment (negarn) [inline src/javascript/app_2/Utils/Date/date-time.js]: can you add a comment what this does

File: src/javascript/_common/base/server_time.js [hunk 5]
```diff
@@ -6,7 +6,7 @@ const ServerTime = (() => {
     let clock_started = false;
     const pending = new PromiseClass();
     let server_time,
-        client_time,
+        perf_request_time,
```
Comment (negarn) [inline src/javascript/_common/base/server_time.js]: IMO better use the full word as perf is not commonly used

**Pull request merged**